admin-notice/conditions/class-wp-version-condition.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Automattic\VIP\Admin_Notice;
+
+class WP_Version_Condition implements Condition {
+
+	private $minimal_version;
+	private $maximum_version;
+
+	/**
+	 * __construct
+	 *
+	 * @param  string $minimal_version - version on which condition should start passing (inclussive)
+	 * @param  string $maximum_version - [optional] version on which condition should no longer pass (exclusive)
+	 * @return WP_Version_Condition
+	 */
+	public function __construct( string $minimal_version, string $maximum_version = null ) {
+		$this->minimal_version = $minimal_version;
+		$this->maximum_version = $maximum_version;
+	}
+
+	public function evaluate() {
+		$current_version = $GLOBALS['wp_version'];
+
+		if ( version_compare( $current_version, $this->minimal_version, '<' ) ) {
```

Comment:
I think we should also allow this to be null, as a typical case will be "show this notice to all WP versions before x.y". By the way, probably "minimum" is a better name than "minimal", although native speakers will know better 😉

Reply:
Updated.